packages/compass-query-bar/src/components/option-editor.tsx

```diff
@@ -148,26 +153,36 @@ const OptionEditor: React.FunctionComponent<OptionEditorProps> = ({
   }, [schemaFields, serverVersion]);
 
   const onFocus = () => {
-    if (hasAutofix && editorRef.current) {
-      if (editorRef.current.editorContents === '') {
+    if (hasAutofix && editorCurrentRef.current) {
+      if (editorCurrentRef.current.editorContents === '') {
         rafraf(() => {
-          editorRef.current?.applySnippet('\\{${}}');
+          editorCurrentRef.current?.applySnippet('\\{${}}');
         });
       }
     }
   };
 
   const onPaste = (event: React.ClipboardEvent<HTMLDivElement>) => {
-    if (hasAutofix && editorRef.current) {
-      const editorContents = editorRef.current.editorContents;
+    if (hasAutofix && editorCurrentRef.current) {
+      const editorContents = editorCurrentRef.current.editorContents;
       const snippet = lenientlyFixQuery(editorContents || '');
       if (snippet) {
-        editorRef.current.applySnippet(snippet);
+        editorCurrentRef.current.applySnippet(snippet);
         event.preventDefault();
       }
     }
   };
 
+  const onBlurEditor = useCallback(() => {
+    if (
+      !!editorCurrentRef.current &&
+      editorCurrentRef.current.editorContents !== editorInitialValueRef.current
+    ) {
+      track('Query Edited', { option_name: name });
```

Comment:
This could potentially be a noisy event, especially if it's picking up the change from empty to the auto supplied `{}`. 
I was under the impression we were more interested if queries that were applied from the ai are edited. Should we only be tracking edits in the situation where the original query was applied from the ai?

Reply:
Chatted about this live.
Amplitude journeys can cover if ai prompt submitted, then query edited (or query run).
Alena mentioned Julia was thinking about a general `Query Edited` event, similar to what we have for aggregations.